h2o-genmodel/src/main/java/hex/genmodel/tools/PredictCsv.java

```diff
@@ -390,6 +394,51 @@ private static void usage() {
     System.exit(1);
   }
 
+  private void checkMissingColumns(final String[] parsedColumnNamesArr) {
+    final String[] modelColumnNames = model.m._names;
+    final Set<String> parsedColumnNames = new HashSet<>(parsedColumnNamesArr.length);
+    for (int i = 0; i < parsedColumnNamesArr.length; i++) {
+      parsedColumnNames.add(parsedColumnNamesArr[i]);
+    }
+
+    List<String> missingColumns = new ArrayList<>();
+    for (String columnName : modelColumnNames) {
+
+      if (!parsedColumnNames.contains(columnName) && !columnName.equals(model.m._responseColumn)) {
+        missingColumns.add(columnName);
+      } else {
+        parsedColumnNames.remove(missingColumns);
+      }
+    }
+    
+    if(missingColumns.size() > 0){
+      final StringBuilder stringBuilder = new StringBuilder("There were ");
+      stringBuilder.append(missingColumns.size());
+      stringBuilder.append(" missing columns found in the input data set: {");
+
+      for (int i = 0; i < missingColumns.size(); i++) {
+        stringBuilder.append(missingColumns.get(i));
+        if(i != missingColumns.size() - 1) stringBuilder.append(",");
+      }
+      stringBuilder.append('}');
+      System.out.println(stringBuilder);
+    }
+    
+    if(parsedColumnNames.size() > 0){
+      final StringBuilder stringBuilder = new StringBuilder("Detected ");
+      stringBuilder.append(parsedColumnNames.size());
+      stringBuilder.append(" unused columns in the input data set: {");
+
+      final Iterator<String> iterator = parsedColumnNames.iterator();
+      while (iterator.hasNext()){
+        stringBuilder.append(iterator.next());
+        if(iterator.hasNext()) stringBuilder.append(",");
+      }
+      stringBuilder.append('}');
+      System.out.println(stringBuilder);
```

Comment:
should probably use Log.info()? (same above)

Reply:
Unfortunately, there is no Log.info here. Had the same reflection as you did.

Reply:
> Thanks for working on this so quickly! :-) Any chance that these can be turned into Python warnings when run from `mojo_predict_pandas` in Python?

Please create another JIRA for that. Raising warnings in Python in real time would require some kind of IPC between the Java `PredictCsv` backend. This  could ultimately make the actual prediction a little bit slower to start. Non-realtime warnings are far easier - I can imagine Python warning you at the end of prediction. There is a lot of potential - we might also introduce a completely new type returned by `mojo_predict_*` methods, providing you with overview of what happend before and during the scoring.

Let's keep this PR's scope to the original request - make the information available in the output.